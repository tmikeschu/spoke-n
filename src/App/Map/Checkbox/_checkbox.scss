.checkboxes {
  display: flex;
  align-items: center;
  @include map-info-border;
  padding: 0 1%;
  justify-content: space-between;
  >article {
    width: 40%;
  }

  @media (max-width: 640px) {
    font-size: .5em;
    flex-direction: column;
    >article {
      width: 100%;
    }
  }
}

