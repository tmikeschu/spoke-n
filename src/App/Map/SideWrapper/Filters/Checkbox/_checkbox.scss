.Checkboxes {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  >article {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    width: 100%;
    article {
      margin: 8px 0;
      width: 50%;
      display: flex;
      align-items: center;
      label {
        width: 100%;
        &:hover, &:focus {
          color: $color2;
          cursor: pointer;
        }
      }
    }
  }
}

