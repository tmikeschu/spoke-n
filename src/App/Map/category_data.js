const categoryIcons = {
  stay: "https://maps.google.com/mapfiles/ms/icons/blue.png",
  checkout: "https://maps.google.com/mapfiles/ms/icons/green.png",
  avoid: "https://maps.google.com/mapfiles/ms/icons/red.png",
  bike_shop: "https://maps.google.com/mapfiles/ms/icons/mechanic.png",
  other: "https://maps.google.com/mapfiles/ms/icons/purple.png",
  endOfDay: "https://maps.google.com/mapfiles/ms/icons/flag.png",
  cycling: "https://maps.google.com/mapfiles/ms/icons/cycling.png",
}

const categoryLabels = {
  "stay": "Places to Stay",
  "checkout": "Cool Spots",
  "avoid": "Places to Avoid",
  "bike_shop": "Bike Shops",
  "other": "Other",
  "": "All",
  "DISPLAYNONE": "None"
};

export { categoryIcons, categoryLabels }

