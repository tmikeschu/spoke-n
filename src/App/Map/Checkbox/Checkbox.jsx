import React, { Component } from 'react';
import { categoryLabels } from '../category_data';

export default class Checkbox extends Component {
  categoryLabels() {
    return categoryLabels;
  }

  render() {
    const category = this.props.category;

    return(
      <article className="checkbox">
        <input 
          type="checkbox"
          value={category}
          onChange={event => this.props.filterPins(event)}
        />
        <label>{ this.categoryLabels()[category] }</label>
      </article>
    );
  }
}

