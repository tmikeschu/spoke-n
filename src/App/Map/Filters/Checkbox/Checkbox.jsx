import React from 'react'
import PropTypes from 'prop-types'
import { categoryLabels } from '../../category_data'

const Checkbox = ({ category, filterPins }) => (
  <article className="checkbox">
    <input 
      type="checkbox"
      value={category}
      onChange={event => filterPins(event)}
    />
    <label>{ categoryLabels[category] }</label>
  </article>
)

Checkbox.propTypes = {
  category: PropTypes.string.isRequired,
  filterPins: PropTypes.func.isRequired
}

export default Checkbox

