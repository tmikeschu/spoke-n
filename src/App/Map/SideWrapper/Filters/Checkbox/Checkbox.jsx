import React from 'react'
import PropTypes from 'prop-types'
import { categoryLabels } from '../../../category_data'

const Checkbox = ({ category, filterPins, pinFilters }) => (
  <article className="checkbox">
    <input 
      type="checkbox"
      value={category}
      onChange={event => filterPins(event)}
      checked={pinFilters.includes(category)}
    />
    <label>{ categoryLabels[category] }</label>
  </article>
)

Checkbox.propTypes = {
  category: PropTypes.string.isRequired,
  filterPins: PropTypes.func.isRequired,
  pinFilters: PropTypes.array.isRequired
}

export default Checkbox

