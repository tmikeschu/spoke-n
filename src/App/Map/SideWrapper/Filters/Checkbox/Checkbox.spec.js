import React from 'react'
import { shallow } from "enzyme"
import Checkbox from "./Checkbox"
import { categoryLabels } from '../../../category_data'

describe("<Checkbox />", () => {
  const props = {
    category: "stay",
    filterPins: function () {},
    pinFilters: []
  }
  const checkbox = shallow(<Checkbox {...props} />)

  it("renders without crashing", () => {
    expect(checkbox).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(checkbox.getNodes()).toMatchSnapshot()
    })
  })
})
