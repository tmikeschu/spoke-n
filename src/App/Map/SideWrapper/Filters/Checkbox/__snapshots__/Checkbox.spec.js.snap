// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> snapshot is valid 1`] = `
<article
  className="checkbox"
>
  <input
    checked={false}
    id="stay"
    onChange={[Function]}
    type="checkbox"
    value="stay"
  />
  <label
    htmlFor="stay"
  >
    Places to Stay
  </label>
</article>
`;
